### PR TITLE
Create VoteConfirmationModal

### PR DESCRIPTION
**Pull request by MiltonTulli**
# Description

Create Vote confirmation modal in proposal to trigger confirmation before send transaction.  
This change is only for UI. Since this issue is for getting the actual voting power for snapshot implementation, will build the logic for this component there https://github.com/DXgovernance/dxvote/issues/746.

Closes https://github.com/DXgovernance/dxvote/issues/786

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

# Checklist:

- [x] My code follows the existing style of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any UI changes have been tested and made responsive for mobile views

# Screenshot:
Screen Shot 2022-04-11 at 20 07 50

### CODE REVIEW

File: src/components/Guilds/ProposalPage/ProposalVoteCard/VoteConfirmationModal/VoteConfirmationModal.styled.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  margin: 0 8px;
+  border: 0;
+  flex: 1;
+  margin-right: 0;
+`;
+
+export const CancelButton = styled(ConfirmButton)`
+  background: transparent;
+  border: 1px solid ${({ theme }) => theme.colors.muted};
+  margin-left: 0;
```
Comment (hhamud) [inline src/components/Guilds/ProposalPage/ProposalVoteCard/VoteConfirmationModal/VoteConfirmationModal.styled.ts]:
```suggestion
export const CancelButton = styled(ConfirmButton).attrs(() => {
variant: "secondary"
})`
  background: transparent;
  border: 1px solid ${({ theme }) => theme.colors.muted};
  margin-left: 0;
 color: transparent;
```

File: src/components/Guilds/ProposalPage/ProposalVoteCard/VoteConfirmationModal/VoteConfirmationModal.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+              <InfoValue grey> {previousVotingPercentage}% </InfoValue>
+              {' -> '} {votingPower}%
+            </InfoValue>
+          </InfoRow>
+        </Widget>
+        <ActionWrapper>
+          <CancelButton
+            variant="secondary"
+            onClick={onDismiss}
+            color="transparent"
```
Comment (hhamud) [inline src/components/Guilds/ProposalPage/ProposalVoteCard/VoteConfirmationModal/VoteConfirmationModal.tsx]: I think we should move these variants into the cancelbutton components shown above therefore this should be like this shown below
```suggestion
          <CancelButton
            onClick={onDismiss}
```

File: src/components/Guilds/ProposalPage/ProposalVoteCard/index.tsx [hunk 3]
```diff
@@ -164,6 +170,14 @@ const ProposalVoteCard = () => {
           </ButtonsContainer>
         )}
       </SidebarCardContent>
+      <VoteConfirmationModal
+        isOpen={modalOpen}
+        onDismiss={() => setModalOpen(false)}
+        onConfirm={confirmVoteProposal}
+        selectedAction="Yes"
+        votingPower="0.12"
+        previousVotingPercentage="0"
```
Comment (hhamud) [inline src/components/Guilds/ProposalPage/ProposalVoteCard/index.tsx]: can we make this more general?

Comment (MiltonTulli) [reply src/components/Guilds/ProposalPage/ProposalVoteCard/index.tsx]: What do you mean with "more general"? This is hardcoded. Dealing with real values on next pr. Changed for `totalLocked` that is more representative fieldName

File: src/components/Guilds/ProposalPage/ProposalVoteCard/VoteConfirmationModal/VoteConfirmationModal.tsx (hunk 2 above)
Comment (MiltonTulli) [reply src/components/Guilds/ProposalPage/ProposalVoteCard/VoteConfirmationModal/VoteConfirmationModal.tsx]: Nice suggestion! Done!

**Review by hhamud — APPROVED** [review]

**Pull request merged**